### PR TITLE
Package errpy.0.0.10

### PR DESCRIPTION
**Pull request by jasontatton**
### `errpy.0.0.10`
Errpy: An Error Recovering Python Parser implemented in Rust
This is a new project to provide an error-recovering Python parser, implemented in Rust based on tree-sitter. Our short-term goal is to use it in the pyre type checker, in order to provide better IDE features.

---
* Homepage: https://github.com/facebook/errpy
* Source repo: git+https://github.com/facebook/errpy.git
* Bug tracker: https://github.com/facebook/errpy/issues

---
:camel: Pull-request generated by opam-publish v2.2.0

### CODE REVIEW

File: packages/errpy/errpy.0.0.10/opam [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+synopsis: "Errpy: An Error Recovering Python Parser implemented in Rust"
+description:
+  "This is a new project to provide an error-recovering Python parser, implemented in Rust based on tree-sitter. Our short-term goal is to use it in the pyre type checker, in order to provide better IDE features."
+maintainer: ["pyre@meta.com"]
+authors: ["pyre@meta.com"]
+license: "MIT"
+homepage: "https://github.com/facebook/errpy"
+bug-reports: "https://github.com/facebook/errpy/issues"
+depends: [
+  "ocaml"
```
Comment (mseri) [inline packages/errpy/errpy.0.0.10/opam:12]:
```suggestion
  "ocaml" {< "5.0"}
```
Due to
```
#=== ERROR while compiling errpy.0.0.10 =======================================#
# context              2.2.0~alpha3~dev | linux/x86_64 | ocaml-base-compiler.5.1.0 | pinned(https://github.com/facebook/errpy/archive/refs/tags/0.0.10.tar.gz)
# path                 ~/.opam/5.1/.opam-switch/build/errpy.0.0.10
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p errpy -j 255 @install
# exit-code            1
# env-file             ~/.opam/log/errpy-7-2037ce.env
# output-file          ~/.opam/log/errpy-7-2037ce.out
### output ###
# File "vendor/ocaml/interop/ocamlrep_ocamlpool/dune", line 8, characters 9-18:
# 8 |   (names ocamlpool)))
#              ^^^^^^^^^
# (cd _build/default/vendor/ocaml/interop/ocamlrep_ocamlpool && /usr/bin/gcc -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -D_FILE_OFFSET_BITS=64 -g -I /home/opam/.opam/5.1/lib/ocaml -o ocamlpool.o -c ocamlpool.c)
# ocamlpool.c: In function 'ocamlpool_reserve_block':
# ocamlpool.c:383:14: error: too few arguments to function 'caml_shared_try_alloc'
#   383 |   value* p = caml_shared_try_alloc(d->shared_heap, wosize, tag, 0);
#       |              ^~~~~~~~~~~~~~~~~~~~~
# In file included from ocamlpool.c:296:
# /home/opam/.opam/5.1/lib/ocaml/caml/shared_heap.h:33:8: note: declared here
#    33 | value* caml_shared_try_alloc(struct caml_heap_state*,
#       |        ^~~~~~~~~~~~~~~~~~~~~
```

Comment (mseri) [inline packages/errpy/errpy.0.0.10/opam:12]:
```suggestion
  "ocaml" {< "5.0"}
```
Due to
```
#=== ERROR while compiling errpy.0.0.10 =======================================#
# context              2.2.0~alpha3~dev | linux/x86_64 | ocaml-base-compiler.5.1.0 | pinned(https://github.com/facebook/errpy/archive/refs/tags/0.0.10.tar.gz)
# path                 ~/.opam/5.1/.opam-switch/build/errpy.0.0.10
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p errpy -j 255 @install
# exit-code            1
# env-file             ~/.opam/log/errpy-7-2037ce.env
# output-file          ~/.opam/log/errpy-7-2037ce.out
### output ###
# File "vendor/ocaml/interop/ocamlrep_ocamlpool/dune", line 8, characters 9-18:
# 8 |   (names ocamlpool)))
#              ^^^^^^^^^
# (cd _build/default/vendor/ocaml/interop/ocamlrep_ocamlpool && /usr/bin/gcc -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -D_FILE_OFFSET_BITS=64 -g -I /home/opam/.opam/5.1/lib/ocaml -o ocamlpool.o -c ocamlpool.c)
# ocamlpool.c: In function 'ocamlpool_reserve_block':
# ocamlpool.c:383:14: error: too few arguments to function 'caml_shared_try_alloc'
#   383 |   value* p = caml_shared_try_alloc(d->shared_heap, wosize, tag, 0);
#       |              ^~~~~~~~~~~~~~~~~~~~~
# In file included from ocamlpool.c:296:
# /home/opam/.opam/5.1/lib/ocaml/caml/shared_heap.h:33:8: note: declared here
#    33 | value* caml_shared_try_alloc(struct caml_heap_state*,
#       |        ^~~~~~~~~~~~~~~~~~~~~
```

File: packages/errpy/errpy.0.0.10/opam [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  "dune" {>= "3.4"}
+  "ppx_deriving"
```
Comment (mseri) [inline packages/errpy/errpy.0.0.10/opam:14]:
```suggestion
  "ppx_deriving" {>= "5"}
```
For
```
#=== ERROR while compiling errpy.0.0.10 =======================================#
# context              2.2.0~alpha3~dev | linux/x86_64 | ocaml-base-compiler.4.04.2 | pinned(https://github.com/facebook/errpy/archive/refs/tags/0.0.10.tar.gz)
# path                 ~/.opam/4.04/.opam-switch/build/errpy.0.0.10
# command              ~/.opam/opam-init/hooks/sandbox.sh build dune build -p errpy -j 255 @install
# exit-code            1
# env-file             ~/.opam/log/errpy-7-225cfd.env
# output-file          ~/.opam/log/errpy-7-225cfd.out
### output ###
# File "dune", line 28, characters 7-24:
# 28 |   (pps ppx_deriving.show))
#             ^^^^^^^^^^^^^^^^^
# Error: Ppx dependency on a non-ppx library "ppx_deriving.show". If
# "ppx_deriving.show" is in fact a ppx rewriter library, it should have (kind
# ppx_rewriter) in its dune file.
# -> required by library "errpy" in _build/default
# -> required by _build/default/META.errpy
# -> required by _build/install/default/lib/errpy/META
# -> required by _build/default/errpy.install
# -> required by alias install
# File "dune", line 28, characters 2-25:
# 28 |   (pps ppx_deriving.show))
#        ^^^^^^^^^^^^^^^^^^^^^^^
# Error: No ppx driver were found. It seems that ppx_deriving.show is not
# compatible with Dune. Examples of ppx rewriters that are compatible with Dune
# are ones using ocaml-migrate-parsetree, ppxlib or ppx_driver.
```

File: packages/errpy/errpy.0.0.10/opam (hunk 1 above)
Comment (jasontatton) [reply packages/errpy/errpy.0.0.10/opam]: Ah ok thanks, I will have re resubmit the PR to resolve that

**Pull request closed without merging**